freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/FreeIpaV1Endpoint.java

```diff
@@ -344,4 +346,13 @@ VerticalScaleResponse verticalScalingByCrn(
     UsedSubnetsByEnvironmentResponse getUsedSubnetsByEnvironment(
             @ValidCrn(resource = CrnResourceDescriptor.ENVIRONMENT) @QueryParam("environmentCrn") String environmentCrn);
 
+    @PUT
+    @Path("rotate_secret")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = ROTATE_SECRETS_BY_CRN, produces = MediaType.APPLICATION_JSON, notes = FreeIpaNotes.FREEIPA_NOTES,
+            nickname = "rotateSecretsByCrn")
+    FlowIdentifier rotateSecretsByCrn(
+            @ValidCrn(resource = CrnResourceDescriptor.ENVIRONMENT) @QueryParam("environment") @NotEmpty String environmentCrn,
```

Comment:
please move crn into request

Reply:
As I saw freeipa endpoint here the crn usually goes into a query param. I would keep it that way because of that.

Reply:
where the is a request object, there is no query param, query param is for get methods as i see

Reply:
There is the changeImageCatalog, verticalScalingByCrn methods for example. Both example has body but they still use environment crn as a query parameter.

Reply:
ok